File: Semestres/11.Febrero-Junio 2022/EstructurasDatosUSB/corte2/Tema21RecorridosArboles/Tree2.py
class Tree:
    def __init__(self,key,left,right):
        self.key = key
        self.left = left
        self.right = right

    def preorder(self):
        salida, leftL, rightL = [], [], []
        salida.append(self.key)
        if(self.left != None):
            leftL = self.left.preorder()
        if(self.right != None):
            rightL = self.right.preorder()

        for l in leftL:
            salida.append(l)

        for r1 in rightL:
            salida.append(r1)


        return salida

    def inororder(self):
        salida, leftL, rightL = [], [], []

        if(self.left != None):
            leftL = self.left.inororder()

        for l in leftL:
            salida.append(l)

        salida.append(self.key)

        if(self.right != None):
            rightL = self.right.inororder()

        for r1 in rightL:
            salida.append(r1)

        return salida

    def posorder(self):
        salida, leftL, rightL = [], [], []

        if(self.left != None):
            leftL = self.left.posorder()

        for l in leftL:
            salida.append(l)

        
        if(self.right != None):
            rightL = self.right.posorder()

        for r1 in rightL:
            salida.append(r1)


        salida.append(self.key)

        return salida
    

File: Semestres/11.Febrero-Junio 2022/EstructurasDatosUSB/corte2/Tema21RecorridosArboles/Tree2.java
import java.util.LinkedList;
import java.util.Arrays;

public class Tree2 {
    private int key;
    private Tree2 left;
    private Tree2 right;

    
    public Tree2(int key, Tree2 left, Tree2 right) {
        this.key = key;
        this.left = left;
        this.right = right;
    }

    

    public int getKey() {
        return key;
    }



    public void setKey(int key) {
        this.key = key;
    }



    public Tree2 getLeft() {
        return left;
    }



    public void setLeft(Tree2 left) {
        this.left = left;
    }



    public Tree2 getRight() {
        return right;
    }



    public void setRight(Tree2 right) {
        this.right = right;
    }



    public LinkedList<Integer> preorder(){
        LinkedList<Integer> list = new LinkedList<Integer>();
        list.add(this.key);
        if(this.left!=null) list.addAll(this.left.preorder());
        if(this.right!=null)list.addAll(this.right.preorder());
        return list;
    }

    public LinkedList<Integer> inorder(){
        LinkedList<Integer> list = new LinkedList<Integer>();
        if(this.left!=null) list.addAll(this.left.inorder());
        list.add(this.key);
        if(this.right!=null)list.addAll(this.right.inorder());
        return list;
    }

    public LinkedList<Integer> posorder(){
        LinkedList<Integer> list = new LinkedList<Integer>();

        if(this.left!=null) list.addAll(this.left.posorder());
        if(this.right!=null)list.addAll(this.right.posorder());
        list.add(this.key);
        return list;
    }
}
